rest/route/project_events.go

```diff
@@ -0,0 +1,99 @@
+package route
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func makeFetchProjectEvents(sc data.Connector) gimlet.RouteHandler {
+	return &projectEventsGet{sc: sc}
+}
+
+type projectEventsGet struct {
+	Timestamp time.Time
+	Limit     int
+	Id        string
+	sc        data.Connector
+}
+
+func (h *projectEventsGet) Factory() gimlet.RouteHandler {
+	return &projectEventsGet{
+		Timestamp: time.Now(),
+		Limit:     10,
+		Id:        "",
+		sc:        h.sc,
+	}
+}
+
+func (h *projectEventsGet) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+
+	h.Id = gimlet.GetVars(r)["project_id"]
+
+	vals := r.URL.Query()
+	k, ok := vals["ts"]
+	if ok && len(k) > 0 {
+		h.Timestamp, err = time.Parse(time.RFC3339, k[0])
+		if err != nil {
+			return errors.Wrap(err, "problem parsing time as RFC-3339")
+		}
+	}
+
+	h.Limit, err = getLimit(vals)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return errors.WithStack(err)
+}
+
+func (h *projectEventsGet) Run(ctx context.Context) gimlet.Responder {
```

Comment:
There needs to be validation here that the request user is either a superuser or a project admin

Reply:
I believe that the middleware should handle this. it has been implemented for other routes.

Reply:
I see a route can be wrapped with a checkUser or superUser where the middleware checks the user is authenticated or a superuser, respectively (rest/route/service.go). 
To implement what John is asking for to also allow a project admin I think I would need to do it here.

Reply:
Yep it should be done here. It should use the checkUser middleware

Reply:
the NewProjectContextMiddleware errors if the project is private and the user isn't an admin. you just have to check user before the project context.

Reply:
Route is wrapped in new middleware